core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -259,6 +264,16 @@ import com.datastax.driver.core.utils.Bytes
         }
       }
 
+      def lookForMissingSeqNr(): Unit = {
+        lookingForMissingSeqNr match {
+          case Some(m) if m.deadline.isOverdue() =>
+            failStage(new IllegalStateException(s"Missing sequence number [$seqNr], got [${m.sawSeqNr}] for " +
+              s"persistenceId [$persistenceId]."))
+          case Some(_) => query(false)
```

Comment:
I think there is a risk of triggering another query before the previous is completed, since `query` can also be called from `continue`. Should probably have a similar check of the `queryState` here as in `continue`.